kernel/clambda.ml

```diff
@@ -839,8 +841,8 @@ let dynamic_int31_compilation fc args =
 
 (* We are relying here on the tags of digits constructors *)
 let digits_from_uint i =
-  let d0 = mkConst_b0 0 in
-  let d1 = mkConst_b0 1 in
+  let d0 = Lint 0 in
+  let d1 = Lint 1 in
```

Comment:
Same here. By the way, for the sake of not repeatedly allocating these values, it might be worth defining them outside the function.